app/views/membership_applications/_search_form.html.haml

```diff
@@ -2,12 +2,12 @@
                                   class: 'form-horizontal' do |f|
   .form-group
     .col-sm-5
-      = f.label :last_name_in,
+      = f.label :user_last_name_in,
                 "#{t('activerecord.attributes.membership_application.last_name')}",
                 class: 'control-label',
                 style: 'text-align: left; font-size: 12px;'
-      = f.collection_select :last_name_in,
-                      MembershipApplication.distinct.order(:last_name).pluck(:last_name),
+      = f.collection_select :user_last_name_in,
+                      User.distinct.order(:last_name).pluck(:last_name).reject(&:blank?),
```

Comment:
This isn't rejecting the MA if `last_name` is blank. It prevents a blank option appearing in the searchfield for lastname (an option that would behave the same as choosing no option at all).

Apart from that: logically, there can't be any MA's for users without first- or lastname.